cmd/checker/main.go

```diff
@@ -53,144 +53,148 @@ func main() {
 	if pckg.Autoupdate == nil {
 		err(ctx, "autoupdate not found")
 		return
 	}
 
-	if pckg.Autoupdate.Source == "npm" {
-		npmVersions := npm.GetVersions(pckg.Autoupdate.Target)
-		if len(npmVersions) == 0 {
-			err(ctx, "no version found on npm")
-			return
-		}
+	// autoupdate exists, download latest versions based on source
+	src := pckg.Autoupdate.Source
+	var versions []version
+	var downloadDir string
+	switch src {
+	case "npm":
+		{
+			// get npm versions and sort
+			npmVersions := npm.GetVersions(pckg.Autoupdate.Target)
+			sort.Sort(npm.ByNpmVersion(npmVersions))
 
-		sort.Sort(npm.ByNpmVersion(npmVersions))
+			// cast to interface
+			for _, v := range npmVersions {
+				versions = append(versions, v)
+			}
 
-		if len(npmVersions) > util.IMPORT_ALL_MAX_VERSIONS {
-			npmVersions = npmVersions[:util.IMPORT_ALL_MAX_VERSIONS]
+			// download into temp dir
+			downloadDir = npm.DownloadTar(ctx, npmVersions[0].Tarball)
 		}
-
-		// print info for the first version
-		firstNpmVersion := npmVersions[0]
+	case "git":
 		{
-			tarballDir := npm.DownloadTar(ctx, firstNpmVersion.Tarball)
-			defer os.RemoveAll(tarballDir) // clean up temp dir
-
-			filesToCopy := pckg.NpmFilesFrom(tarballDir)
+			// make temp dir and clone
+			packageGitDir, direrr := ioutil.TempDir("", src)
+			util.Check(direrr)
+			out, cloneerr := packages.GitClone(ctx, pckg, packageGitDir)
+			if cloneerr != nil {
+				err(ctx, fmt.Sprintf("could not clone repo: %s: %s\n", cloneerr, out))
+				return
+			}
 
-			if len(filesToCopy) == 0 {
-				errormsg := ""
-				errormsg += fmt.Sprintf("No files will be published for version %s.\n", firstNpmVersion.Version)
+			// get git versions and sort
+			gitVersions := git.GetVersions(ctx, pckg, packageGitDir)
+			sort.Sort(git.ByGitVersion(gitVersions))
 
-				for _, filemap := range pckg.NpmFileMap {
-					for _, pattern := range filemap.Files {
-						errormsg += fmt.Sprintf("[Click here to debug your glob pattern `%s`](%s).\n", pattern, makeGlobDebugLink(pattern, tarballDir))
-					}
-				}
-				err(ctx, errormsg)
-				goto moreNpmVersions
+			// cast to interface
+			for _, v := range gitVersions {
+				versions = append(versions, v)
 			}
 
-			fmt.Printf("```\n")
-			for _, file := range filesToCopy {
-				fmt.Printf("%s\n", file.To)
-			}
-			fmt.Printf("```\n")
+			// set download dir
+			downloadDir = packageGitDir
 		}
-
-	moreNpmVersions:
-		// aggregate info for the few last version
-		fmt.Printf("\n%d last versions:\n", util.IMPORT_ALL_MAX_VERSIONS)
+	default:
 		{
-			for _, version := range npmVersions {
-				tarballDir := npm.DownloadTar(ctx, version.Tarball)
-				filesToCopy := pckg.NpmFilesFrom(tarballDir)
-
-				fmt.Printf("- %s: %d file(s) matched", version.Version, len(filesToCopy))
-				if len(filesToCopy) > 0 {
-					fmt.Printf(" :heavy_check_mark:\n")
-				} else {
-					fmt.Printf(" :heavy_exclamation_mark:\n")
-				}
-			}
+			panic(fmt.Sprintf("unknown autoupdate source: %s", src))
```

Comment:
this should be an error, while panic will make the CI fail it won't show as an annotation

Reply:
good catch, committed fix